src/sources/internal_logs.rs

```diff
@@ -64,26 +65,33 @@ async fn run(
         .chain(tokio_stream::wrappers::BroadcastStream::new(
             subscription.receiver,
         ))
+        .filter_map(|log| {
+            future::ready(log.ok().and_then(|l| {
+                if l.contains("internal") {
+                    None
+                } else {
+                    Some(l)
+                }
```

Comment:
Yeah, I am wondering if it'd be feasible to not have this component log at all rather than try to filter its own logs given this is likely to be error prone with string matching. As Bruce mentioned, this would drop any logs that _happen_ to have `internal` in them. This might mean having separate internal events for this component that don't emit logs.

Reply:
Agreed. This feels brittle. I'd rather avoid the use of strings with undocumented side effects if we can help it.

I think it's worth raising this point in core, to suggest one of:

1. Avoiding emitting logs from this source entirely.
2. Having a separate mechanism for logging.
3. Keeping logging, but figuring out a less brittle way to do this than string matching.

I'm also concerned about the cost of matching against every event. As we've seen with matching in Pipelines filtering (albeit primarily regex), the costs add up.

Reply:
I'm ok with not monitoring the `BytesReceived` and `EventsReceived` events for this source entirely.
I'd consider batching them maybe 🤔 (but how 🤷)...
But we should also take a look at the events generated by the transforms and sinks downstream like mentioned [here](https://github.com/vectordotdev/vector/pull/11034#discussion_r795523957), every log sent to a following transform will generate a metric log, and will keep feeding this vicious circle. 🤔

Reply:
Yeah, that is a definite issue if the whatever `internal_logs` is connected to, itself, generates 1 or more log per incoming event. That footgun already exists though, before this change, so I think, as long as we can avoid `internal_logs` generating logs as part of this change that would be sufficient for now.

Curiously users haven't run into that issue in practice, that I'm aware of. Probably only trace-level logging typically results in one log message per event. It is something we should think about for sure though, thanks for raising it up.